features/networking/multus-ipv6.feature

```diff
@@ -50,3 +50,70 @@ Feature: Multus-CNI ipv6 related scenarios
+      | ["spec"]["config"]        | '{ "cniVersion": "0.3.0", "name": "whereabouts", "type": "macvlan", "mode": "bridge", "ipam": { "type": "whereabouts", "range": "fd00:dead:beef:1::1-fd00:dead:beef:1::4/64", "exclude": ["fd00:dead:beef:1::2/128"] } }'|
+    Then the step should succeed
+
+    # Create the three pods which consumes the custom resource
+    Given I obtain test data file "networking/multus-cni/Pods/IPv6/ipv6-pod.yaml"
+    When I run oc create over "ipv6-pod.yaml" replacing paths:
+      | ["spec"]["nodeName"] | "<%= cb.target_node %>" |
+    Then the step should succeed
+    When I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :describe client command with:
+      | resource | pod |
+    Then the output should contain:
+      | fd00:dead:beef:1::1 |
+      | fd00:dead:beef:1::3 |
+      | fd00:dead:beef:1::4 |
+    And the output should not contain:
+      | fd00:dead:beef:1::2 |
+    """
+
+    # Create new net-attach-def via cluster admin to exclude address list
+     Given I obtain test data file "networking/multus-cni/NetworkAttachmentDefinitions/whereabouts-excludeIP.yaml"
+    When I run oc create as admin over "whereabouts-excludeIP.yaml" replacing paths:
+      | ["metadata"]["name"]      | whereabouts-excludeipv6-list                                                                                                                                                                                                                                                                                  |
+      | ["metadata"]["namespace"] | <%= project.name %>                                                                                                                                                                                                                                                                                           |
+      | ["spec"]["config"]        | '{ "cniVersion": "0.3.0", "name": "whereabouts", "type": "macvlan", "mode": "bridge", "ipam": { "type": "whereabouts", "range": "fd00:dead:beef:1::10-fd00:dead:beef:1::16/64", "exclude": ["fd00:dead:beef:1::10/128","fd00:dead:beef:1::11/128","fd00:dead:beef:1::12/128", "fd00:dead:beef:1::13/128"] } }'|
+    Then the step should succeed
+
+    # Create the three pods which consumes the custom resource
+   Given I obtain test data file "networking/multus-cni/Pods/IPv6/ipv6-pod.yaml"
+    When I run oc create over "ipv6-pod.yaml" replacing paths:
+      | ["spec"]["nodeName"]                                                           | "<%= cb.target_node %>"      |
+      | ["metadata"]["name"]                                                           | test-pod-list                |
+      | ["spec"]["template"]["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeipv6-list |
+    Then the step should succeed
+    When I wait up to 30 seconds for the steps to pass:
```

Comment:
@kedark3 No problem, I do the same when I learn cucushift automation. Above step is not really wait 30 seconds, it means that the script will give total 30 seconds to let below steps pass. It may wait for 20 or 25 seconds which depending on the cluster environment. Sometimes it's very difficult to give exactly time. This estimate hard code time is acceptable in some scripts.

Reply:
@weliang1 thanks for that info. In my previous team we used `Wait_for(TaskX, Timeout=XXSeconds)` python based function from https://pypi.org/project/wait-for/ project. Is it correct if I say, in your code, the value 30 second is same as timeout from my example?

Reply:
@kedark3 I guess they are same things. cucushif call it "I wait up to 30 seconds"

Reply:
Thanks @weliang1

Reply:
FYI, from the step name it is `wait up to xxx seconds`, so that the time out value.  The loop will exit once the check within the enclosing step evaluate to `true`

Reply:
@kedark3 I'm trying to introduce jitter for `wait_for` here  https://github.com/openshift/verification-tests/pull/1649